rmc/utils/constraint.py

```diff
@@ -1969,6 +1973,120 @@ def _annotate_gene_ids(ht: hl.Table) -> hl.Table:
     )
 
 
+def check_for_overlapping_intervals(interval_ht: hl.Table, coord_ht: hl.Table) -> None:
+    """
+    Check for overlapping intervals within the same transcript.
+
+    :param interval_ht: RMC HT keyed by RMC interval.
+    :param coord_ht: RMC HT keyed by either RMC interval start or end coordinate.
+    :return: None
+    """
+    coord_ht = coord_ht.annotate(
+        interval_matches=interval_ht.index(coord_ht.locus, all_matches=True),
+    )
+    coord_ht = coord_ht.checkpoint(
+        f"{TEMP_PATH_WITH_FAST_DEL}/rmc_coord_check.ht", overwrite=True
+    )
+    coord_ht = coord_ht.filter(hl.len(coord_ht.interval_matches) > 1)
+
+    # Keep only rows where start coordinate was found in two intervals **in the same transcript**
+    coord_ht = coord_ht.annotate(
+        interval_matches=coord_ht.interval_matches.filter(
+            lambda x: x.transcript == coord_ht.transcript
+        )
+    )
+    coord_ht = coord_ht.filter(hl.len(coord_ht.interval_matches) > 1)
+    n_overlap = coord_ht.count()
+    if n_overlap > 0:
+        coord_ht.show()
```

Comment:
Remove?

Reply:
Oh ok - I thought it may have been a copy paste error from a notebook but if this is something you want to/can output for a check, sounds good!